### PR TITLE
Emc/chore/1493-user-fetch-own-all-listings

### LINKED ISSUE #1493: [projects] Frontend code changes for CM contract v2 support
The underlying assumption for the Marketplace page and a Project Details page is that listings that are either expired or invalid are not shown.

when a user views their own profile, they can see expired listings and edit them.

See related API issues: #1489 & #1490 

# Remaining Todo
 * [x] The User Profile frontend needs to pass expiresAfter=0 when the user is connected *and viewing their own profile*. This would likely be passed down via the useFetchUser hook like `useFetchUser(args, { expiresAfter: "0" })`
 * [x] Update the edit listing logic to support mumbai (see other `actions` for examples)

### PR DESCRIPTION
**Pull request by 0xemc**
## Description

Resolves #1493

### CODE REVIEW

File: carbonmark/hooks/useFetchUser.ts [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-export const useFetchUser = (address?: string, options?: Opts) => {
-  const { network = "polygon", ...swrConfig } = options || {};
-  const { data, ...rest } = useSWR<User>(
-    address
-      ? `${urls.api.users}/${address}?type=wallet&network=${network}`
-      : null,
-    fetcher,
-    swrConfig
+  const { data, ...rest } = useSWR(
+    `${urls.api.users}/${args.params.walletOrHandle}&network=${args.query?.network}`,
```
Comment (Atmosfearful) [inline carbonmark/hooks/useFetchUser.ts:16]: This string is used by SWR as a key to track changes, so we need to:
* [ ] make sure the key changes only when a param changes
* [ ] make sure the fallback key matches the one provided to the SWRConfig in `components/pages/Users/index.tsx`. (Create a shared util to construct both?)
* [ ] make sure address is always lower case (I noticed on production that we aren't consistent here)

**Review by Atmosfearful — CHANGES_REQUESTED** [review]

File: carbonmark/hooks/useFetchUser.ts [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-  network?: "mumbai" | "polygon";
-};
+type GetUserFnT = ClientT["/users/{walletOrHandle}"]["get"];
+
+export const useFetchUser = (
+  args: Parameters<GetUserFnT>[0],
+  options?: SWRConfiguration
+) => {
+  const fetchUser = async () =>
+    (await client["/users/{walletOrHandle}"].get(args)).json();
```
[the remark this answers is not in the document]
Comment (0xemc) [reply carbonmark/hooks/useFetchUser.ts:13]: Fair point, added as an additional option

[the remark this answers is not in the document]
Comment (0xemc) [reply carbonmark/hooks/useFetchUser.ts:13]: Good point, I've set the fetchFn on the client to be the originally fetcher (meta i know..)

File: carbonmark/hooks/useFetchUser.ts (hunk 1 above)
Comment (0xemc) [reply carbonmark/hooks/useFetchUser.ts]: Added a temporary solution to this via a key gen function. A more solid solution is in the works.

**Review by Atmosfearful — APPROVED** [review]

**Review by Atmosfearful — APPROVED** [review]

**Pull request merged**